### ISSUE TITLE
Unable to reproduce source free domain adaptation results

### ISSUE THREAD

**Issue by ujwal-k03** [zhu-xlab/UniDA#3]
Hi, the MA with source data yields results close to what is claimed by the paper. However, the MA without source data yields extremely off and unstable results.

Here are the final accuracy scores for RSSCN->UCM source free MA:
```
[0.58, 0.09, 0.09, 0.59, 0.44, 0.3535714285714286]
	tensor([[0.3573]])
```

Some things that I have tried:

1. Using source data instead of synthetic data in the SDG-MA/main.py file. This seems to alleviate the problem and the results are identical to the MA with source data.
2. Double checking the loss functions and using the generator in eval mode.

I have attached an image that is the synthetic data generated. Please let me know if this is what is to be expected.

Lastly, here is my configuration:
```
python==3.11.5
cuda==12.2
torch==2.0.1
```
![images-100](https://github.com/zhu-xlab/UniDA/assets/86528667/810e55ac-ecdc-4857-b617-0496215330f7)

Any sort of help would be appreciated to rectify my reproduction, will appreciate any clarifications. Many thanks.

**Comment by QingsongXu123** [comment]
Hi, Thanks for your attention and reproduction of our work!
According to the results you have reproduced so far, the core problem is that there are some problems with the generated data and distributions.
Firstly, to reproduce our results, please use the pre-trained model on RSSCN provided by us (link is here: https://drive.google.com/file/d/1EPX0qbq0kqanXDed4iy2kDhreguv-qJz/view?usp=drive_link).
Secondly, please check your data generation part, the training is likely insufficient. If you have further questions about the training details, you can contact us by email: qingsong.xu@tum.de.
Best,
Qingsong

**Comment by ujwal-k03** [comment]
> Hi, Thanks for your attention and reproduction of our work! According to the results you have reproduced so far, the core problem is that there are some problems with the generated data and distributions. Firstly, to reproduce our results, please use the pre-trained model on RSSCN provided by us (link is here: https://drive.google.com/file/d/1EPX0qbq0kqanXDed4iy2kDhreguv-qJz/view?usp=drive_link). Secondly, please check your data generation part, the training is likely insufficient. If you have further questions about the training details, you can contact us by email: [qingsong.xu@tum.de](mailto:qingsong.xu@tum.de). Best, Qingsong

I have sent you a mail. Kindly reply at the earliest. Many thanks.
